src/components/PostDetails.tsx

```diff
@@ -1,106 +1,129 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment, CommentData } from '../types/Comment';
+import { addComment, deleteComment, getComments } from '../api/api';
 
-export const PostDetails: React.FC = () => {
-  return (
-    <div className="content" data-cy="PostDetails">
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
+type Props = {
+  activePost: Post;
+};
 
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
-        </div>
+export const PostDetails: React.FC<Props> = ({ activePost }) => {
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+  const [isLoadingBtn, setIsLoadingBtn] = useState(false);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [isCommentFormVisible, setIsCommentFormVisible] = useState(false);
 
-        <div className="block">
-          <Loader />
+  useEffect(() => {
+    setIsLoadingComments(true);
+    setIsCommentFormVisible(false);
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+    getComments(activePost.id)
+      .then(setComments)
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoadingComments(false));
+  }, [activePost]);
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+  const handleDeleteComment = (commentId: number) => {
+    setComments(commentsList =>
+      commentsList.filter(comment => comment.id !== commentId),
+    );
 
-          <p className="title is-4">Comments:</p>
+    deleteComment(commentId)
+      .then(() => {})
+      .catch(() => {
+        setComments(commentsList => [
+          ...commentsList,
+          comments.find(comment => comment.id === commentId)!,
+        ]);
+        setIsError(true);
+      });
```

Comment:
The `deleteComment` function should handle errors more gracefully. Consider providing user feedback or retry logic instead of just setting the error state.